### ISSUE TITLE
Nested gitignores do not seem to work 

### ISSUE THREAD

**Issue by TryerGit** [streetsidesoftware/vscode-spell-checker#2434]
Hello,

I have begun using this extension along with my LaTeX files. Consider the following structure:
```
/outerfolder/
       main.tex
       .gitignore
       /innerfolder/
            main.tex
            .gitignore
```

`main.tex` in both parent and child folders can be:

```tex
\documentclass[11pt]{article}
\begin{document}
	
Theorem

Theorm

\end{document}
```

`.gitignore` of `/outerfolder/` is:

```
/innerfolder/*
!.gitignore

```

`.gitignore` of `/innerfolder/` is:

```
!main.tex
!.gitignore
```

Essentially, the child folder's `.gitignore` should force tracking of the child folder's `main.tex`

This can be verified by running:

`/innerfolder$ git check-ignore -v main.tex`

which results in :

`outerfolder/innerfolder/.gitignore:1:!main.tex       main.tex`

Opening `main.tex` of `/outerfolder/` works fine with the extension correctly catching the spelling error in `Theorm`.  However, opening `main.tex` of `/innerfolder/` does not trigger spell check. One obtains:

```
main.tex
File is exclude by .gitignore. Line: 1, Glob: /innerfolder/*
outerfolder/.gitignore
File is NOT spell checked because it is not in the workspace.

```

Any help in supporting nested gitignores will be greatly appreciated.

Thank you.

**Comment by Jason3S** [comment]
@TryerGit,

It is a missing feature: https://github.com/streetsidesoftware/cspell/issues/3530

**Comment by TryerGit** [comment]
Ah okay. @Jason3S No problem. I have been able to make a setting change based on your other messages on the same topic elsewhere so that it works bypassing all gitignores.

**Comment by TryerGit** [comment]
I would imagine that correct parsing of `git check-ignore -v`  output is rather nontrivial..
